src/Blueprint/Blueprint/Cmdlets/BlueprintCmdletBase.cs

```diff
@@ -0,0 +1,382 @@
+
+                graphRbacManagementClient.TenantID = DefaultProfile.DefaultContext.Tenant.Id.ToString();
+
+                return graphRbacManagementClient;
+            }
+            set => graphRbacManagementClient = value;
+        }
+
+        /// <summary>
+        /// Authorization client
+        /// </summary>
+        private IAuthorizationManagementClient authorizationManagementClient;
+
+        public IAuthorizationManagementClient AuthorizationManagementClient
+        {
+            get
+            {
+                return authorizationManagementClient = authorizationManagementClient ?? AzureSession.Instance.ClientFactory.CreateArmClient<AuthorizationManagementClient>(DefaultProfile.DefaultContext, AzureEnvironment.Endpoint.ResourceManager);
+ 
+            }
+            set => authorizationManagementClient = value;
+        }
+
+        /// <summary>
+        /// ARM client
+        /// </summary>
+        private IResourceManagementClient resourceManagerClient;
+        public IResourceManagementClient ResourceManagerClient
+        { 
+            get
+            {
+                return resourceManagerClient = resourceManagerClient ?? new ResourceManagementClient(
+                                                   DefaultProfile.DefaultContext.Environment.GetEndpointAsUri(AzureEnvironment.Endpoint.ResourceManager),
+                                                   ClientCredentials);
+            }
+            set => resourceManagerClient = value;
+        }
+
+        #endregion Properties
+
+        #region Cmdlet Overrides
+        protected override void WriteExceptionError(Exception ex)
+        {
+            var aggEx = ex as AggregateException;
+
+            if (aggEx != null && aggEx.InnerExceptions != null)
+            {
+                foreach (var e in aggEx.Flatten().InnerExceptions)
+                {
+                    WriteExceptionError(e);
+                }
+
+                return;
+            }
+
+            throw ex;
+        }
+        #endregion Cmdlet Overrides
+
+        #region Protected Methods
+        /// <summary>
+        /// Check if an assignment with the same name exists
+        /// </summary>
+        /// <param name="scope"> Subscription scope</param>
+        /// <param name="name"> Assignment name</param>
+        /// <param name="invertCheck"> Inverts the check logic from check for exists to check for not-exits </param>
+        protected void CheckIfAssignmentExist(string scope, string name, bool invertCheck)
+        {
+            PSBlueprintAssignment existingAssignment = null;
+
+            try
+            {
+                existingAssignment = BlueprintClient.GetBlueprintAssignment(scope, name);
+            }
+            catch (Exception ex)
+            {
+                if (ex is CloudException cex && cex.Response.StatusCode != System.Net.HttpStatusCode.NotFound)
+                {
+                    throw new CloudException(cex.Message);
```

Comment:
>cex.Message [](start = 45, length = 11)

This comes back out as a localized message, but where do we configure the language we want on the client?

Reply:
We talked about this offline. PowerShell doesn't support localization.

---
In reply to: [257093171](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257093171) [](ancestors = 257093171)